### ISSUE TITLE
Trailing comma from ScalaGenerator when using schemaVersionProvider

### ISSUE THREAD

**Issue by robbytx** [jOOQ/jOOQ#5649]
Changes in [3.8.0](https://github.com/jOOQ/jOOQ/commit/8145f77f2e53829b51941f5826fae2188369e0e6#diff-aaa83edc55e463bc44fecee3fc51fa3eR4431) have caused jOOQ to generate broken Scala code due to Scala's intolerance of trailing commas in array definitions:
```
/**
 * This class is generated by jOOQ.
 */
@Generated(
  value = Array(
    "http://www.jooq.org",
    "jOOQ version:3.8.6",
    "schema version:1",   //<- here is the trailing comma 
  ),
  date = "2016-11-07T02:58:49.390Z",
  comments = "This class is generated by jOOQ"
)
class DefaultCatalog extends CatalogImpl("") {
...
```

This leads to compilation errors:
```
[ERROR] ~/server/src/main/generated/jooq/DefaultCatalog.scala:34: error: illegal start of simple expression
[ERROR]   ),
[ERROR]   ^
[ERROR] ~/server/src/main/generated/jooq/DefaultCatalog.scala:55: error: ')' expected but eof found.
[ERROR] }
[ERROR]  ^
```

This is a regression from 3.7.x.  Unfortunately, one of my projects also need [enum support](https://github.com/jOOQ/jOOQ/issues/4427), so I've had to switch back to JavaGenerator and go to other lengths to workaround this.

**Comment by lukaseder** [comment]
Thanks for pointing this out. Indeed, this seems to have slipped through our integration tests, as we're currently not using any `<schemaVersionProvider>` with the scala integration tests

**Comment by lukaseder** [comment]
A workaround would of course be to turn off the schema version provider functionality, in case of which you won't run into this issue. The drawback is that the code is re-generated every time (although unmodified files will still not be touched)

**Comment by lukaseder** [comment]
Fix will be backported to 3.8.7 (https://github.com/jOOQ/jOOQ/issues/5651)

**Comment by robbytx** [comment]
Thanks for the quick turnaround!
